extensions/common/gcp/gcp-core/src/test/java/org/eclipse/dataspaceconnector/gcp/core/iam/IamServiceImplTest.java

```diff
@@ -0,0 +1,156 @@
+import static org.mockito.AdditionalMatchers.and;
+import static org.mockito.Answers.RETURNS_SMART_NULLS;
+import static org.mockito.Mockito.any;
+import static org.mockito.Mockito.argThat;
+import static org.mockito.Mockito.doNothing;
+import static org.mockito.Mockito.doThrow;
+import static org.mockito.Mockito.eq;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+class IamServiceImplTest {
+
+    private final String projectId = "test-project-Id";
+    private final String serviceAccountName = "test-service-account";
+    private final String serviceAccountEmail = String.format("%s@%s.iam.gserviceaccount.com", serviceAccountName, projectId);
+    private final String serviceAccountDescription = "service-account-description";
+    private IamService iamApi;
+    private IAMClient iamClient;
+    private IamCredentialsClient iamCredentialsClient;
+    private ServiceAccountWrapper testServiceAccount;
+
+    private static ApiException apiExceptionWithStatusCode(StatusCode.Code code) {
+        return ApiExceptionFactory.createException(
+                new Exception(), new StatusCode() {
+                    @Override
+                    public Code getCode() {
+                        return code;
+                    }
+
+                    @Override
+                    public Object getTransportCode() {
+                        return null;
+                    }
+                }, false);
+    }
+
+    @BeforeEach
+    void setUp() {
+        var monitor = Mockito.mock(Monitor.class);
+        iamClient = Mockito.mock(IAMClient.class, RETURNS_SMART_NULLS);
+        iamCredentialsClient = Mockito.mock(IamCredentialsClient.class, RETURNS_SMART_NULLS);
+        testServiceAccount = new ServiceAccountWrapper(serviceAccountEmail, serviceAccountName, serviceAccountDescription);
+        iamApi = IamServiceImpl.Builder.newInstance(monitor, projectId)
+                .iamClientSupplier(() -> iamClient)
+                .iamCredentialsClientSupplier(() -> iamCredentialsClient)
+                .build();
+    }
+
+    @Test
+    void testCreateServiceAccount() {
+        var serviceAccount = ServiceAccount.newBuilder().setEmail(serviceAccountEmail).build();
+        when(iamClient.createServiceAccount(any(CreateServiceAccountRequest.class))).thenReturn(serviceAccount);
+
+        ServiceAccountWrapper createdServiceAccount = iamApi.getOrCreateServiceAccount(serviceAccountName, serviceAccountDescription);
+        assertThat(createdServiceAccount.getEmail()).isEqualTo(serviceAccountEmail);
+        assertThat(createdServiceAccount.getDescription()).isEqualTo(serviceAccountDescription);
+    }
+
+    @Test
+    void testCreateServiceAccountThatExistsAlreadyWithMatchingDescription() {
+        var serviceAccount = ServiceAccount.newBuilder().setEmail(serviceAccountEmail).setDescription(serviceAccountDescription).build();
```

Comment:
please clearly separate with empty lines the three steps: arrange, act and assert